python/mscclpp_benchmark/allreduce.cu

```diff
@@ -775,3 +776,67 @@ extern "C" __global__ void __launch_bounds__(1024, 1)
     globalFlag += 1;
   }
 }
+
+// -------------------------------------------
+// AllReduce6
+// NVLS
+// -------------------------------------------
+
+#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ >= 900
+#define MULTIMEM_ST(val, ptr)                                                                                   \
+  asm volatile("multimem.st.global.v4.f32 [%0], {%1,%2,%3,%4};" ::"l"(ptr), "r"(val.x), "r"(val.y), "r"(val.z), \
+               "r"(val.w)                                                                                       \
+               : "memory");
+// specific PTX for fp16 reduction. bf16 would be multimem.ld_reduce.global.add.v4.bf16x2 etc
+#define MULTIMEM_LD(val, ptr)                                     \
+  asm("multimem.ld_reduce.global.add.v4.f32 {%0,%1,%2,%3}, [%4];" \
+      : "=r"(val.x), "=r"(val.y), "=r"(val.z), "=r"(val.w)        \
+      : "l"(ptr)                                                  \
+      : "memory");
+
```

Comment:
Do we need to move this to mscclpp source, and provide version for different types